src/stirling/source_connectors/socket_tracer/protocols/common/always_contiguous_data_stream_buffer_impl.cc

```diff
@@ -263,14 +263,41 @@ std::map<size_t, size_t>::const_iterator AlwaysContiguousDataStreamBufferImpl::G
   return iter;
 }
 
-std::string_view AlwaysContiguousDataStreamBufferImpl::Get(size_t pos) const {
+void AlwaysContiguousDataStreamBufferImpl::EnforceTimestampMonotonicity(size_t pos,
+                                                                        size_t chunk_end) {
+  // Get timestamp for chunk which is <= pos.
+  auto it = timestamps_.upper_bound(pos);
+  if (it == timestamps_.begin()) {
+    return;
+  }
+  --it;
+
+  // Loop from chunk_start up to but not including chunk_end.
+  // The next element after chunk_end should not be part of the contiguous block.
+  prev_timestamp_ = 0;
+  for (; it != timestamps_.end() && it->first < chunk_end; ++it) {
+    if (prev_timestamp_ > 0 && it->second < prev_timestamp_) {
+      LOG(WARNING) << "For chunk pos " << it->first << ", "
+                   << "detected non-monotonically increasing timestamp " << it->second
+                   << ". Adjusting to previous timestamp + 1: " << prev_timestamp_ + 1;
```

Comment:
I realized we missed this on the previous review, but our [style guide](https://github.com/pixie-io/pixie/blob/main/styleguide/cc_style.md#formatted-strings) disallows `<<` for string formatting. We should use `absl::Substitute` here instead.